### PR TITLE
Dajung

### PR DESCRIPTION
**Pull request by ss01363**

### CODE REVIEW

**Comment by cvanderw** [comment]
Great to see your pull request! Could you assign some reviewers for the pull request?

**Comment by cvanderw** [comment]
@insunj Looks like @ss01363 made a lot of great changes here in the client. Care to take a look at her pull request?

**Comment by cvanderw** [comment]
@insunj please review when you get a chance. Thanks!

**Comment by cvanderw** [comment]
@amyjang sorry, I meant this one!

**Comment by ss01363** [comment]
oops sorry T.T I've been off and forgot to assign reviewers...

File: TreasureHunt/Assets/2.Scripts/Hunter/FieldView/TreasureSetupController.cs [hunk 1]
```diff
@@ -4,64 +4,68 @@
...
-
-	}
-
-	void Start(){
-		if (!gameObject.GetComponent<NetworkManager> ().enabled) {
-			GetGameTreasure ("gg");		
+		} 
+		else if (instance != this) 
+		{
+			Destroy (gameObject);
```
Comment (cvanderw) [inline TreasureHunt/Assets/2.Scripts/Hunter/FieldView/TreasureSetupController.cs:20]: Is this the style for calling functions in C#?

The examples I see in [the style guide](https://msdn.microsoft.com/en-us/library/ff926074.aspx) see to suggest `Destroy(gameObject)` as the suggested way.

Here and elsewhere in your code.

File: TreasureHunt/Assets/2.Scripts/Hunter/FieldView/TreasureSetupController.cs [hunk 2]
```diff
@@ -4,64 +4,68 @@
...
 	}
 
 	//infos included in json:
 	//	flag,
-	//	game_id, game_name, treasure_count, maker_id, status,
+	//	game_id, game_name, treasure_count, maker_id, status, participant
 	//	treasure_id, treasure_name, description, game_id, location, point, catchgame_cat, target_img_url
 	public GameObject game;
 	public GameObject treasure;
 	public static List<string> m_Data = new List<string>();
```
Comment (cvanderw) [inline TreasureHunt/Assets/2.Scripts/Hunter/FieldView/TreasureSetupController.cs]: Why use "m_" prefix for variables?

File: TreasureHunt/Assets/2.Scripts/Hunter/FieldView/TreasureSetupController.cs [hunk 3]
```diff
@@ -73,14 +77,15 @@ void Awake()
...
 				Debug.Log ("something wrong with treasure_count");
+				return gameTreasurePanel;
 			}
-			// make treasures
-			for (int j = 0; j < treasures.Count; j++) {
+			// make treasure objects 
+			for (int j = 0; j < treasures.Count; j++) 
+			{
 				string treasure_id = treasures [j] ["treasure_id"];
 				MakeNewTreasure (newGame, treasures[j]["treasure_id"], treasures[j]["treasure_name"], treasures[j]["destination"],
```
Comment (cvanderw) [inline TreasureHunt/Assets/2.Scripts/Hunter/FieldView/TreasureSetupController.cs]: Some of these lines look too long. Keep to about 100 char columns.

File: TreasureHunt/Assets/2.Scripts/Hunter/GameListView/PopupController.cs [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+		// make popup
+		GameObject popup = null;
+		if (isMyGame) 
+		{
+			popup = (GameObject)Resources.Load ("MyGamePopup");
+		} 
+		else 
+		{
+			popup = (GameObject)Resources.Load ("SearchGamePopup");
+			Debug.Log ("wow");
```
Comment (cvanderw) [inline TreasureHunt/Assets/2.Scripts/Hunter/GameListView/PopupController.cs:83]: Wow? ;)

File: TreasureHunt/Assets/2.Scripts/Hunter/GameListView/PopupController.cs [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+				for (int j = 0; j < treasures.Count; j++) 
+				{
+					// todo: 
+				}
+			}
+		}
+	}
+
+	public void TurnDownPopup(){
+		Debug.Log ("called");
```
Comment (cvanderw) [inline TreasureHunt/Assets/2.Scripts/Hunter/GameListView/PopupController.cs]: Is this using spaces or tabs? https://msdn.microsoft.com/en-us/library/ff926074.aspx#Anchor_1

File: TreasureHunt/Assets/2.Scripts/Hunter/GameListView/PopupController.cs [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+			NetworkManager.instance.SendData (str);
+		}
+		// reset data so that if you go back to Field View, treasure setup would start again
+		TreasureSetupController.userGameTreasureData = ""; // needed if we use cache flush
+		// close popup and delete this gamelist from scroll view
+		Destroy(GameObject.Find("Canvas/Scroll View/Viewport/Content/" + gameID));
+		TurnDownPopup();
+	}
+
+	/*
```
Comment (cvanderw) [inline TreasureHunt/Assets/2.Scripts/Hunter/GameListView/PopupController.cs]: What happening with this dead code? Do you need it?

File: TreasureHunt/Assets/2.Scripts/Hunter/Player/MockMovement.cs [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+	public GameObject player;
+
+	IEnumerator DoIt(){
+		var jsonData = JSON.Parse (data);
+		var locs = jsonData ["Locs"];
+
+		UserMovementController con = player.GetComponent<UserMovementController>();
+
+		for (int i = 0; i < locs.Count; i++) {
+			yield return new WaitForSeconds (1.5f);
```
Comment (cvanderw) [inline TreasureHunt/Assets/2.Scripts/Hunter/Player/MockMovement.cs:30]: What does this do?

**Review by cvanderw — COMMENTED** [review]
I left some basic comments. I don't want to hold back your work though.

File: TreasureHunt/Assets/2.Scripts/Hunter/FieldView/TreasureSetupController.cs (hunk 2 above)
Comment (ss01363) [reply TreasureHunt/Assets/2.Scripts/Hunter/FieldView/TreasureSetupController.cs]: Oh those lines were to list out DB components. DB labels are wrote in that style! code was not wrote in that style :) would that be a problem..?

File: TreasureHunt/Assets/2.Scripts/Hunter/GameListView/PopupController.cs (hunk 5 above)
Comment (ss01363) [reply TreasureHunt/Assets/2.Scripts/Hunter/GameListView/PopupController.cs]: It's tabs!

File: TreasureHunt/Assets/2.Scripts/Hunter/GameListView/PopupController.cs (hunk 6 above)
Comment (ss01363) [reply TreasureHunt/Assets/2.Scripts/Hunter/GameListView/PopupController.cs]: Uhm... no.... thought that maybe more modulizing would be good but would not gonna use that i guess

Comment (ss01363) [reply TreasureHunt/Assets/2.Scripts/Hunter/GameListView/PopupController.cs]: Deleted it!

File: TreasureHunt/Assets/2.Scripts/Hunter/Player/MockMovement.cs (hunk 7 above)
Comment (ss01363) [reply TreasureHunt/Assets/2.Scripts/Hunter/Player/MockMovement.cs:30]: That script is for mocking movements of players so it gets locations and move players in some time interval!

File: TreasureHunt/Assets/2.Scripts/Hunter/FieldView/TreasureSetupController.cs (hunk 2 above)
Comment (cvanderw) [reply TreasureHunt/Assets/2.Scripts/Hunter/FieldView/TreasureSetupController.cs]: Oh, I'm specifically talking about the List<string> data member. The data type is written as `m_Data`. Why not just `data`?

File: TreasureHunt/Assets/2.Scripts/Hunter/GameListView/PopupController.cs (hunk 5 above)
Comment (cvanderw) [reply TreasureHunt/Assets/2.Scripts/Hunter/GameListView/PopupController.cs]: Ah. I believe the convention in the style guide is to configure the editor to use spaces for the tabs. But(!) it's more important that the team is consistent. Again something to agree upon with the C# code.

File: TreasureHunt/Assets/2.Scripts/Hunter/Player/MockMovement.cs (hunk 7 above)
Comment (cvanderw) [reply TreasureHunt/Assets/2.Scripts/Hunter/Player/MockMovement.cs:30]: Thanks for the details. Good to know.

File: TreasureHunt/Assets/2.Scripts/Hunter/FieldView/TreasureSetupController.cs (hunk 1 above)
Comment (ss01363) [reply TreasureHunt/Assets/2.Scripts/Hunter/FieldView/TreasureSetupController.cs:20]: ummm do you mean space in between destroy and (gameobject)? the editor automatically changed it T.T

File: TreasureHunt/Assets/2.Scripts/Hunter/FieldView/TreasureSetupController.cs (hunk 2 above)
Comment (ss01363) [reply TreasureHunt/Assets/2.Scripts/Hunter/FieldView/TreasureSetupController.cs]: oh i see! I dont know... that line would have come from one of 유정's code i guess... not used in the script so deleted that line! Thanks :)

**Comment by hyunSo** [comment]
Dear mentors @cvanderw @amyjang ,
We are in hurry so I am going to merge it...

**Pull request merged**